baseapp/params.go

```diff
@@ -108,3 +110,10 @@ func WithHTTPServer(server *http.Server) Param {
 		return nil
 	}
 }
+
+func WithTLSConfig(tlsConfig *tls.Config) Param {
+	return func(s *Server) error {
+		s.server.TLSConfig = tlsConfig
```

Comment:
`s.server` is `nil` when `Param` functions are called, unless `WithHTTPServer` appears earlier in the parameter list.